Prevent Groovy illegal access warnings

oehme opened the pull request:
Start all processes that are likely to use Groovy with the
necessary --add-opens arguments to avoid noisy warnings that
our users can't do anything about.

Fixes #6725

oehme commented:
I'm not convinced we should do this for `Test` - The list of illegally accessed packages there is getting longer and longer as I try to run our own test suite.

marcphilipp commented on subprojects/base-services/src/main/java/org/gradle/internal/jvm/GroovyJpmsWorkarounds.java:
```diff
@@ -0,0 +1,34 @@
+    /**
+     * These JVM arguments should be passed to any process that will be using Groovy on Java 9+ to avoid noisy illegal access warnings that the user can't do anything about.
+     */
+    public static final List<String> SUPPRESS_COMMON_GROOVY_WARNINGS = Collections.unmodifiableList(Arrays.asList(
```
Can we use `ImmutableList.of()` here?

oehme replied:
Nope, we don't want to load Guava in the command line client.

oehme commented:
We could even decide to pass `--illegal-access=deny` to the daemon and tell users to add the relevant `--add-opens` if a plugin wants do do illegal accesses. But maybe that's too early and should be done in 6.0, when more libraries out there are fixed.

oehme commented:
@marcphilipp I've removed the add-opens for `Test` tasks and cut the list of add-opens down to the minimum needed for most Groovy build scripts and GroovyCompile. I've adjusted our own Test tasks to deny illegal access and add the relevant opens clauses. This is what users would have to do as well, with the add-opens depending on their code.

oehme commented:
Even trying to only open up the things our tests really need proves to be quite a lot - There are many illegal acceses currently hidden behind the one-shot warning: https://builds.gradle.org/viewQueued.html?itemId=16496243

marcphilipp commented:
Indeed. 😲

oehme commented:
I've fixed the most obvious one (the TAPI doing an illegal access). CodeNarc is broken because Groovy is broken. The illegal accesses in the embedded executer (which also affects TestKit in debug mode) are harder to get rid of. The illegal accesses in the file tree tests are just in the test code and thus less of a problem, but we should probably still find a better way, e.g. by forking another process to test them.

wolfs commented:
@oehme I reverted your change since it caused CI breakages: https://builds.gradle.org/viewLog.html?buildId=16582372&buildTypeId=Gradle_Check_PerformanceTestCoordinator
